### PR TITLE
feat: add checkbox component

### PR DESCRIPTION
**Pull request by MaximeJineMaraval**
## Changes description 🧑‍💻

Implement new component : Checkbox ✅ 
- Add the new component's module
- Reference the new component in the Vitamin theme
- Add the new component in the sample app
- Update the sample app to manage the "reset" button
- Update the README.md

## Context 🤔

https://github.com/Decathlon/vitamin-android/issues/4

## Checklist ✅

 - [x] I have reviewed the submitted code
 - [x] I have tested on a phone device/emulator
 - [x] I have tested on a tablet device/emulator
 - [x] I have tested on a kiosk device/emulator

#### Phone

Screenshot_1616508568
Screenshot_1620116008

#### Tablet

Screenshot_1616508539
Screenshot_1620116051

#### Kiosk

Screenshot_1616508651
Screenshot_1620116074

Screenshot_1616508672
Screenshot_1620116115

### CODE REVIEW

File: checkboxes/build.gradle [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
+}
+
+dependencies {
+    api(project(":foundation"))
+    api(deps.material)
```
Comment (GerardPaligot) [inline checkboxes/build.gradle:26]: Should not be necessary. `deps.material` dependency is already defined on `foundation` module with `api` call to be accessible for all module consumers.

File: checkboxes/src/main/res/values/styles-checkboxes.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <style name="Widget.Vitamin.Checkbox" parent="Widget.MaterialComponents.CompoundButton.CheckBox">
+        <item name="android:textColor">@color/vtmn_checkbox_text</item>
+        <item name="buttonTint">@color/vtmn_checkbox_box</item>
```
Comment (GerardPaligot) [inline checkboxes/src/main/res/values/styles-checkboxes.xml:6]: Maybe add the text appearance too (body2)? If there is any change in material library, we'll stay with the expected text appearance in our design system.

**Review by GerardPaligot — CHANGES_REQUESTED** [review]

File: checkboxes/build.gradle (hunk 1 above)
Comment (MaximeJineMaraval) [reply checkboxes/build.gradle]: Updated ✅ 
I also change it in the button module

File: checkboxes/src/main/res/values/styles-checkboxes.xml (hunk 2 above)
Comment (MaximeJineMaraval) [reply checkboxes/src/main/res/values/styles-checkboxes.xml:6]: Done ✅ 

I also add the disabled checked state (you can see it in the screenshots)

**Review by GerardPaligot — APPROVED** [review]

**Review by charlottemougenot — APPROVED** [review]

**Pull request merged**